tests/h/views/api_auth_test.py

```diff
@@ -15,6 +17,106 @@
 from h.views import api_auth as views
 
 
+@pytest.mark.usefixtures('routes', 'user_service', 'oauth_service')
+class TestOAuthAuthorizeController(object):
+    def test_get_verifies_client_id(self, auth_ctrl, pyramid_request,
```

Comment:
I've renamed and more importantly, also added checks that the authority of the client matches `request.authority`.